src/lib/falcor/FalcorRouter.js

```diff
@@ -11,25 +28,169 @@ export default class FalcorRouter extends BaseRouter.createClass([
+        });
+        query += "&fields=slug"
+        pathSet[2].forEach((field, index) => {
+          if (field !== 'slug') {
+            query += "," + ghostify(field);
+          }
+        })
+        query += "&limit=" + pathSet.slugs.length.toString();
+        ghostArticleQuery(query).then((data) => {
+          data = data.posts;
+          const results = [];
+          data.forEach((article, index) => {
+            pathSet[2].forEach((field) => {
+              let ghostField = ghostify(field);
+              results.push({
+                path: ["articlesBySlug", article.slug, field],
+                value: article[ghostField]
+              });
+            });
+          });
+          resolve(results);
+        });
+      });
+    }
+  },
+  {
+    //Get custom article data from SQL database
+    route: "articlesBySlug[{keys:slugs}]['issue', 'category', 'description']",
+    get: (pathSet) => {
+      return new Promise((resolve, reject) => {
+        let query = {slug: {$in: pathSet.slugs}};
+        let projection = {_id: 0};
+        pathSet[2].forEach((field) => {
+          projection[field] = 1;
+        });
+        projection['slug'] = 1;
+        SQLArticleQuery(query, projection).then((data) => {
+          const results = [];
+          data.forEach((article) => {
+            pathSet[2].forEach((field) => {
+              results.push({
+                path: ["articlesBySlug", article.slug, field],
+                value: article[field]
+              });
+            });
+          });
+          resolve(results);
         });
+      });
+    }
+  },
+  {
+    // Get author information from article
+    route: "articlesBySlug[{keys:slugs}]['authors'][{integers:indices}]",
+    get: (pathSet) => {
+      console.log("here");
+      return new Promise((resolve, reject) => {
+        console.log("there");
+        let query = {slug: {$in: pathSet.slugs}};
+        let projection = {_id: 0, slug: 1, authors: 1};
+        SQLArticleQuery(query, projection).then((data) => {
+          console.log("made it and data coming here");
+          const results = [];
+          console.log(data);
+          data.forEach((article) => {
+            let authors = article.authors;
+            let indices = pathSet.indices;
+            for (let i = 0; i < indices.length; i++) {
+              let index = indices[i];
+              if (index >= authors.length) {
+                continue;
+              }
+              results.push({
+                path: ['articlesBySlug', article.slug, 'authors', index],
+                value: $ref(['authorsBySlug', authors[index]])
+              });
+            }
+          });
+          console.log("resolving");
+          console.log(results);
+          resolve(results);
+        });
+      });
+    }
+  },
+  {
+    // Get articles in chronological order by a consecutive range of indices where both min and max are multiples
```

Comment:
I'll address all your concerns about just making it a page as a part of the path here instead of rewriting many times:

So my reasons for doing it like this is:
1. Like this you can easily change the page size you are querying for, as you decide the range you call
2. Falcor only allows to return one item per path, so otherwise we would return an atom, which I guess we could also do, but I thought we were avoiding that.

I'm very open to doing it differently though, but I think it's quite valid concerns though, especially the pageLength part. We could of course do something like articlesByPage[(integer pageLength)][(integer pageNumber)] if we wanted though. but that wouldn't be an obvious path either.